magic_the_gathering/turn.py

```diff
@@ -0,0 +1,42 @@
+from magic_the_gathering.game_state import GameState
+from magic_the_gathering.phases.combat_beginning import CombatBeginningPhase
+from magic_the_gathering.phases.combat_damage import CombatDamagePhase
+from magic_the_gathering.phases.combat_declare_attackers import CombatDeclareAttackersPhase
+from magic_the_gathering.phases.combat_declare_blockers import CombatDeclareBlockersPhase
+from magic_the_gathering.phases.combat_end import CombatEndPhase
+from magic_the_gathering.phases.draw import DrawPhase
+from magic_the_gathering.phases.end import EndPhase
+from magic_the_gathering.phases.main import MainPhase
+from magic_the_gathering.phases.untap import UntapPhase
+from magic_the_gathering.phases.upkeep import UpkeepPhase
+
+
+class Turn:
+    def __init__(self):
+        self.__phases = [
+            UntapPhase(),
+            UpkeepPhase(),
+            DrawPhase(),
+            MainPhase(name="Main Phase 1"),
+            CombatBeginningPhase(),
+            CombatDeclareAttackersPhase(),
+            CombatDeclareBlockersPhase(),
+            CombatDamagePhase(),
+            CombatEndPhase(),
+            MainPhase(name="Main Phase 2"),
+            EndPhase(),
+        ]
+        # TODO: Add a logger
+
+    def run(self, game_state: GameState) -> GameState:
+        for phase in self.__phases:
+            game_state = phase.run(game_state)
+            # TODO: Here is the logic of a turn below, how can we implement it using the Phase classes efficiently?
+            # TODO: players_get_priority seems to condition the behavior of the phase, should we consider having a different parent class for these?
```

Comment:
I like the idea of having `Phase` and the child `PhaseWerePlayersGetPriority`

Reply:
And I guess they would have this special method you called `player_having_priority_may_take_an_action`